components/client/src/test/unit/java/com/hotels/styx/client/loadbalancing/strategies/RoundRobinStrategyTest.java

```diff
@@ -39,15 +42,17 @@
 
 public class RoundRobinStrategyTest {
     private static final StubConnectionFactory connectionFactory = new StubConnectionFactory();
-    private static final ConnectionPool POOL_1 = createConnectionPoolFor("localhost", 1);
-    private static final ConnectionPool POOL_2 = createConnectionPoolFor("localhost", 2);
-    private static final ConnectionPool POOL_3 = createConnectionPoolFor("localhost", 3);
+    private static final RemoteHost POOL_1 = createConnectionPoolFor("localhost", 1);
+    private static final RemoteHost POOL_2 = createConnectionPoolFor("localhost", 2);
+    private static final RemoteHost POOL_3 = createConnectionPoolFor("localhost", 3);
+    private static Origin origin;
 
-    private static SimpleConnectionPool createConnectionPoolFor(String host, int port) {
-        return new SimpleConnectionPool(
-                newOriginBuilder(host, port).build(),
+    private static RemoteHost createConnectionPoolFor(String host, int port) {
+        origin = newOriginBuilder(host, port).build();
+
+        return remoteHost(origin, new SimpleConnectionPool(origin,
```

Comment:
The arguments to `remoteHost` would be easier to understand if they were on separate lines.